contracts/rewards/src/state.rs

```diff
@@ -0,0 +1,383 @@
+use std::collections::HashMap;
+
+use cosmwasm_schema::cw_serde;
+use cosmwasm_std::{Addr, Storage, Uint256};
+use cw_storage_plus::{Item, Map};
+
+use crate::{error::ContractError, msg::RewardsParams};
+
+#[cw_serde]
+pub struct Config {
+    pub params: RewardsParams,
+}
+
+#[cw_serde]
+pub struct EpochTally {
+    pub epoch_num: u64,
+    pub contract: Addr,
+    pub total_events: u64,
+    pub participation: HashMap<Addr, u64>,
+    pub rewards_rate: Uint256,
```

Comment:
This is the rewards that should be distributed for this entire epoch? In that case, I'll drop the word "rate" in there just to not confused with the per block reward rate thing.

Reply:
No this is the rewards rate that was in effect during that epoch. We need this in case the rewards rate changes for future epochs, but the rewards themselves haven't been distributed.